src/features/sign-up/ui/SignUpForm.tsx

```diff
@@ -34,14 +47,35 @@ const SignUpForm: FC<Props> = props => {
     onSubmitSuccess: data => {
       executeIfOnline(() => signUp(data));
     },
+    criteriaMode: 'all',
+    shouldUseNativeValidation: false,
```

Comment:
This is because of the `zodResolver` function as you can see [here](https://github.com/ChildMindInstitute/mindlogger-app-refactor/blob/df1a0ea0fc577da2e0756e703d6830405946d540/src/shared/lib/validation/zodResolver.ts#L93), by removing it or setting it to `true` will result in not retrieve all errors at once.